core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/upgrade/upgrade.yml

```
@@ -35,6 +35,27 @@
   set_fact:
     pg_primary_node: "{{ pg_primary_node_found.stdout | default('') }}"
 
+- name: PostgreSQL | Query custom tablespaces
+  become_user: postgres
+  postgresql_query:
+    query: "select pg_tablespace_location(oid) as location from pg_tablespace where spcname not in ('pg_default', 'pg_global')"
+  delegate_to: "{{ pg_primary_node }}"
+  run_once: true
+  register: pg_custom_tablespaces
+
+- name: PostgreSQL | Set 'pg_custom_tablespace_locations' fact
+  block:
+    - name: PostgreSQL | Initialize 'pg_custom_tablespace_locations' fact
+      set_fact:
+        pg_custom_tablespace_locations: [ ]
+
+    - name: PostgreSQL | Set 'pg_custom_tablespace_locations' fact
+      set_fact:
+        pg_custom_tablespace_locations: "{{ pg_custom_tablespace_locations + [pg_tablespace.location] }}"
```

You can consider combining these two tasks into one using  `default([])`:
```suggestion
        pg_custom_tablespace_locations: "{{ pg_custom_tablespace_locations | default([]) + [pg_tablespace.location] }}"
```

---

If the query result contains 0 elements, this fact will be undefined and this will require to add a default in a few other places

---

Then `pg_custom_tablespaces.query_result | map(attribute='location') | reject('==', '')` is an option instead of `loop`.

---

modified